src/ast/semantic_analyser.cpp

```diff
@@ -695,7 +706,16 @@ void SemanticAnalyser::visit(Binop &binop)
       buf << "with '" << rhs << "'";
       bpftrace_.error(err_, binop.loc, buf.str());
     }
-
+    // TODO: need to emit a warning here. Introduce a warning stream
+    // and refactor bpftrace_.error to accept a level?
+    else if (lhs == Type::integer && rhs == Type::integer
+        && binop.left->type.is_signed != binop.right->type.is_signed) {
+      std::cerr << "WARNING: Sign mismatch for '" << opstr(binop) << "'" << std::endl;
```

Comment:
This is bugged, should only print a warning for comparison of different types a bitwise and or a bitshift isn't impacted by this.